backends/p4tools/testgen/core/small_step/extern_stepper.cpp

```diff
@@ -0,0 +1,949 @@
+             result->emplace_back(nextState);
+         }},
+        /* ======================================================================================
+         *  drop_and_exit
+         *  This internal extern drops the entire packet and exits.
+         *  We do this by clearing the packet variable and pushing an exit continuation.
+         * ======================================================================================
+         */
+        {"*.drop_and_exit",
+         {},
+         [this](const IR::MethodCallExpression* /*call*/, const IR::Expression* /*receiver*/,
+                IR::ID& /*methodName*/, const IR::Vector<IR::Argument>* /*args*/,
+                const ExecutionState& state, SmallStepEvaluator::Result& result) {
+             auto* nextState = new ExecutionState(state);
+             // If the drop variable is tainted, we also mark the port tainted.
+             if (state.hasTaint(programInfo.dropIsActive())) {
+                 nextState->set(programInfo.getTargetOutputPortVar(),
+                                programInfo.createTargetUninitialized(
+                                    programInfo.getTargetOutputPortVar()->type, true));
+             }
+             nextState->add(new TraceEvent::Generic("Packet marked dropped."));
+             nextState->setProperty("drop", true);
+             nextState->replaceTopBody(Continuation::Exception::Drop);
+             result->emplace_back(nextState);
+         }},
+        /* ======================================================================================
+         *  copy_in
+         *  Copies values from @param srcRef to @param targetParam following the
+         *  copy-in/copy-out semantics of P4. We use this function to copy values in and out of
+         *  individual program pipes.
+         *  @param direction signifies the qualified class of the targetParam ("in", "inout", "out",
+         *  or "<none>").
+         *  All parameters that have direction "out" are set uninitialized.
+         * ======================================================================================
+         */
+        {"*.copy_in",
+         {"srcRef", "targetParam", "direction", "forceUninitialized"},
+         [this](const IR::MethodCallExpression* /*call*/, const IR::Expression* /*receiver*/,
+                IR::ID& /*methodName*/, const IR::Vector<IR::Argument>* args,
+                const ExecutionState& state, SmallStepEvaluator::Result& result) {
+             const auto* globalRef = args->at(0)->expression;
+             if (!(globalRef->is<IR::Member>() || globalRef->is<IR::PathExpression>())) {
+                 TESTGEN_UNIMPLEMENTED("Global input %1% of type %2% not supported", globalRef,
+                                       globalRef->type);
+             }
+
+             const auto* paramRef = args->at(1)->expression;
+             if (!(paramRef->is<IR::Member>() || paramRef->is<IR::PathExpression>())) {
```

Comment:
this is a bit confusing, an argument is not a parameter. perhaps the error message needs to be changed. this is a problem below as well.